### PR TITLE
Bug fix in polynomial

### PR DESCRIPTION
**Pull request by nishnik**
@chenchfort @isuruf

### CODE REVIEW

**Comment by certik** [comment]
What bug is it fixing (you should explain that in the git commit log)? You should also write a test that fails in master and is fixed in this PR.

**Comment by chenchfort** [comment]
Yes, looks like it should be `var` instead of `a.get_var()`.

**Comment by nishnik** [comment]
I will write the tests. And should I also write a function for :
```
RCP<const Symbol> var = symbol("");
if (a.get_var()->get_name() == "") {
    var = b.get_var();
} else if (b.get_var()->get_name() == "") {
    var = a.get_var();
} else if (!(a.get_var()->__eq__(*b.get_var()))) {
    throw std::runtime_error("Error: variables must agree.");
} else {
    var = a.get_var();
}
```
It has been used many times in the code.

**Comment by abhinavagarwal07** [comment]
I think this PR should restrict to correcting the bug and adding corresponding test case only.

**Comment by nishnik** [comment]
@chenchfort
We are never going to get this:
`a.get_var()->get_name() == "" `
Because this check is already in `is_canonical`.
Why are we using it then?
Correct me if I am wrong.

**Comment by chenchfort** [comment]
@nishnik 
`is_canonical` allows variable with empty string, provided that the polynomial's dictionary is empty (which would make it 0) or the size of dictionary is 1 (a constant term).

File: symengine/tests/basic/test_polynomial.cpp [hunk 1]
```diff
@@ -64,13 +64,19 @@ TEST_CASE("Negative of a UnivariateIntPolynomial", "[UnivariateIntPolynomial]")
...
+
     const UnivariateIntPolynomial a(x, 2, std::move(adict_));
     const UnivariateIntPolynomial b(x, 2, std::move(bdict_));
+    const UnivariateIntPolynomial c(none, 0, std::move(cdict_));
 
-    RCP<const Basic> c = sub_poly(b, a);
-    REQUIRE(c->__str__() == "3*x**2 + x + 1");
+    RCP<const Basic> d = sub_poly(b, a);
+    REQUIRE(d->__str__() == "3*x**2 + x + 1");
+    d = sub_poly(c,a);
```
Comment (isuruf) [inline symengine/tests/basic/test_polynomial.cpp]: Space after comma

**Comment by CodeMaxx** [comment]
Looks good to me.

**Comment by isuruf** [comment]
You should remove the binary file `symengine/tests/basic/test_sets` from the history

**Comment by nishnik** [comment]
Sorry! My bad! 
Fixed @isuruf

**Pull request merged**